chiselFrontend/src/main/scala/chisel3/Bitwise.scala

```diff
@@ -0,0 +1,140 @@
+   * @group Bitwise
+   */
+  def << (that: Int): T
+
+  /** @group SourceInfoTransformMacro */
+  def do_<< (that: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T
+
+  /** Dynamic left shift operator
+   *
+   * @param that a hardware component
+   * @return this $coll dynamically shifted left by `that` many places, shifting in zeros from the right
+   * @note The width of the returned $coll is `width of this + pow(2, width of that) - 1`.
+   * @group Bitwise
+   */
+  def << (that: UInt): T
+
+  /** @group SourceInfoTransformMacro */
+  def do_<< (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T
+
+  /** Static right shift operator
+   *
+   * @param that an amount to shift by
+   * @return this $coll with `that` many least significant bits truncated
+   * $unchangedWidth
+   * @group Bitwise
+   */
+  // REVIEW TODO: redundant
+  def >> (that: BigInt): T
+
+  /** @group SourceInfoTransformMacro */
+  def do_>> (that: BigInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T
+
+  /** Static right shift operator
+   *
+   * @param that an amount to shift by
+   * @return this $coll with `that` many least significant bits truncated
+   * $unchangedWidth
+   * @group Bitwise
+   */
+  def >> (that: Int): T
+
+  /** @group SourceInfoTransformMacro */
+  def do_>> (that: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T
+
+  /** Dynamic right shift operator
+   *
+   * @param that a hardware component
+   * @return this $coll dynamically shifted right by the value of `that` component, inserting zeros into the most
+   * significant bits.
+   * $unchangedWidth
+   * @group Bitwise
+   */
+  def >> (that: UInt): T
+
+  /** @group SourceInfoTransformMacro */
+  def do_>> (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T
+
+  // TODO: refactor to share documentation with Num or add independent scaladoc
+  /** Unary negation (expanding width)
+   *
+   * @return a $coll equal to zero minus this $coll
+   * $constantWidth
+   * @group Arithmetic
+   */
+  final def unary_- (): T = macro SourceInfoTransform.noArg
+
+  /** Unary negation (constant width)
+   *
+   * @return a $coll equal to zero minus this $coll shifted right by one.
+   * $constantWidth
+   * @group Arithmetic
+   */
+  final def unary_-% (): T = macro SourceInfoTransform.noArg
+
+  override def do_< (that: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool =
+    compop(sourceInfo, LessOp, that)
+  override def do_> (that: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool =
+    compop(sourceInfo, GreaterOp, that)
+  override def do_<= (that: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool =
+    compop(sourceInfo, LessEqOp, that)
+  override def do_>= (that: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool =
+    compop(sourceInfo, GreaterEqOp, that)
```

Comment:
Thanks for your comment about this before. I see the issue with `compop` being defined in `Bits`. I'm trying to reason about this, but I think that's wrong.

It would seem more natural if this PR could be reduced to only add `Bitwise[T <: Data]` (a bitwise analog to `Num[T <: Data]`). However, `pad` requires `binop`...

Let me bring this up at the developer meeting today. I think there may be a prerequisite PR that sorts out where `binop`, etc. should live first before this goes through.